modules/gojira-core/src/main/java/com/flipkart/gojira/core/BindingsModule.java

```diff
@@ -38,8 +40,15 @@ protected void configure() {
           new ProfileOrTestMethodInterceptor();
       bindInterceptor(
           Matchers.any(),
-          Matchers.annotatedWith(ProfileOrTest.class),
+          new NoCompilerMethodMatcher().and(Matchers.annotatedWith(ProfileOrTest.class)),
           profileOrTestMethodInterceptor);
     }
   }
+
+  private static class NoCompilerMethodMatcher extends AbstractMatcher<Method> {
+    @Override
+    public boolean matches(Method method) {
+      return !method.isSynthetic() && !method.isBridge();
```

Comment:
does this include lambdas as well?

Reply:
No, it don't think it does. 
Reference: https://medium.com/datadriveninvestor/java-synthetic-methods-what-are-these-52f77efb347a.